src/Strategy/Cargo.hs

```diff
@@ -156,6 +201,69 @@ instance ToJSON CargoProject
 instance AnalyzeProject CargoProject where
   analyzeProject _ = getDeps
 
+data CargoPackage = CargoPackage
+  { unLicense :: Maybe Text.Text
+  , unLicenseFile :: Maybe Text.Text
+  }
+  deriving (Eq, Show)
+
+cargoPackageCodec :: TomlCodec CargoPackage
+cargoPackageCodec =
+  CargoPackage
+    <$> dioptional (Toml.text "license") .= unLicense
+    <*> dioptional (Toml.text "license-file") .= unLicenseFile
+
+-- |Representation of a Cargo.toml file. See
+-- [here](https://doc.rust-lang.org/cargo/reference/manifest.html)
+-- for a description of this format.
+newtype CargoToml = CargoToml
+  {cargoPackage :: CargoPackage}
+  deriving (Eq, Show)
+
+cargoTomlCodec :: TomlCodec CargoToml
+cargoTomlCodec = diwrap (Toml.table cargoPackageCodec "package")
+-- ^^ The above is a bit obscure. It's generating a TomlCodec CargoPackage and
+-- then using 'diwrap'/Coercible to make a TomlCodec CargoToml.  I can't use
+-- 'CargoToml <$>' because TomlCodec aliases (Codec a a) and only (Codec a)
+-- has a Functor instance, so I'd end up with a (Codec CargoPackage CargoToml).
+
+instance LicenseAnalyzeProject CargoProject where
+  licenseAnalyzeProject = analyzeLicenses . cargoToml
+
+-- |Analyze a Cargo.toml for license information. The format is documented
+-- (here)[https://doc.rust-lang.org/cargo/reference/manifest.html#the-license-and-license-file-fields]
+analyzeLicenses :: (Has ReadFS sig m, Has Diagnostics sig m) => Path Abs File -> m [LicenseResult]
+analyzeLicenses tomlPath = do
+  CargoPackage
+    { unLicenseFile = maybeLicenseFile
+    , unLicense = maybeLicense
+    } <-
+    cargoPackage <$> readContentsToml cargoTomlCodec tomlPath
```

Comment:
[nit]: This `do` syntax is really confusing to read.  It took me way too long to realize that this is where `maybeLicense` and `maybeLicenseFile` are _defined_, rather than _referenced_, which is what I originally thought.

I think we should use more obvious naming/syntax here (feel free to adjust this however you like it):
```haskell
pkg <- cargoPackage <$> readContentsToml cargoTomlCodec tomlPath
```

Reply:
Please let me know if this is any easier.